### ISSUE TITLE
[BUG] Texts not visible in dark mode

### ISSUE THREAD

**Issue by debangshu919** [CipherYuvraj/Algorithm-Visualiser-Platform#11]
### Description
In dark mode, certain texts are not visible. This impacts accessibility and user experience.

### Steps to reproduce
1.  Go to the deployed landing page: [Algorithm Vizualizer Platform](https://algorithm-visualiser-yuvraj.onrender.com/)
2. Switch to `dark` mode
3. Observe the hero section and other affected areas where the text blends into the background.

### Screenshots

<img width="1907" height="986" alt="Image" src="https://github.com/user-attachments/assets/02ce2e78-0598-4b22-ac44-59c026690b2c" />

<img width="1910" height="987" alt="Image" src="https://github.com/user-attachments/assets/661d7104-c1d8-4e34-a04d-470fe87c8992" />

### Expected behviour
- Text should adapt to dark mode by switching to white or light-colored text with sufficient contrast.

**Comment by debangshu919** [comment]
i would like to work on this issue
